modules/core/src/main/java/org/apache/ignite/internal/sql/SyncResultSetAdapter.java

```diff
@@ -107,39 +106,32 @@ public T next() {
     private static class IteratorImpl<T> implements Iterator<T> {
         private AsyncResultSet<T> curRes;
 
-        private CompletionStage<? extends AsyncResultSet<T>> nextPageStage;
-
         private Iterator<T> curPage;
 
         IteratorImpl(AsyncResultSet<T> ars) {
             curRes = ars;
-
-            advance();
+            curPage = ars.currentPage().iterator();
```

Comment:
`SyncResultSetAdapter` is used by embedded client also.

I suggest to keep prefetching for embedded client (turn it off for thin client only).

@korlov42 what do you think about this?